### PR TITLE
upgrade spdlog to 1.8.5

### PR DESCRIPTION
**Pull request by rongou**
The new version includes `fmt` version 7.1.3 which fixed the problem with cuda 10.x.

@harrism @kkraus14

### CODE REVIEW

**Review by kkraus14 — APPROVED** [review]

**Review by kkraus14 — CHANGES_REQUESTED** [review]
Actually, do we want to leave the pinning open ended like this? Would we be better off just pinning to a specific version like 1.8.2 for now?

**Comment by rongou** [comment]
@kkraus14 Looks like we have to first upgrade the integration repo, so it's a two step process. I'd prefer we keep rolling forward.

**Review by harrism — APPROVED** [review]

**Review by mike-wendt — CHANGES_REQUESTED** [review]
Please update with pinnings that are decided on in rapidsai/integration#198

**Comment by mike-wendt** [comment]
rerun tests

**Review by mike-wendt — APPROVED** [review]
Approved, but needs to wait until rapidsai/integration#198 is merged with the conda pkg fix detailed in that PR

**Comment by mike-wendt** [comment]
> Approved, but needs to wait until [rapidsai/integration#198](https://github.com/rapidsai/integration/pull/198) is merged with the conda pkg fix detailed in that PR

Until then the testing in this PR will continue to fail due to the upstream conda-forge changes

**Review by harrism — APPROVED** [review]

**Comment by rongou** [comment]
rerun tests

**Comment by rongou** [comment]
rerun tests

**Comment by kkraus14** [comment]
@rongou it looks like what's happening is that when building the `rmm` Python conda package it's not using the local `librmm` conda package that was built with `spdlog 1.8.2` for some reason.

@raydouglass could this be related to the recent CI build changes?

**Comment by rongou** [comment]
@kkraus14 I fixed an issue with the spdlog conda package, but somehow the build script is not picking up the latest version from conda-forge but getting it from `pkgs/main` instead. Any ideas why?

**Comment by kkraus14** [comment]
rerun tests

**Comment by kkraus14** [comment]
It looks like the a100 build didn't kick back off for some reason, but seems like everything checks out now.

**Review by raydouglass — DISMISSED** [review]
Requesting changes to prevent merging. Since `spdlog` 1.8.2 requires gcc 9.3, that means `rmm` would then require gcc 9.3 which is not conda solveable with the rest of RAPIDS.

**Comment by raydouglass** [comment]
> It looks like the a100 build didn't kick back off for some reason, but seems like everything checks out now.

A100 Optional tests are now disabled for Project Flash. A100 testing is done as part of `gpuCI/rmm/gpu/cuda/11.0/python/3.8/ubuntu18.04` now.

**Comment by kkraus14** [comment]
> Requesting changes to prevent merging. Since `spdlog` 1.8.2 requires gcc 9.3, that means `rmm` would then require gcc 9.3 which is not conda solveable with the rest of RAPIDS.

Yup, we'll have to solve the greater solving issue before we merge this.

**Comment by harrism** [comment]
Moving to 0.19

**Comment by kkraus14** [comment]
rerun tests

**Comment by rongou** [comment]
@kkraus14 do we still want to do this? Actually `spdlog` has some new versions now, I can update this PR to the new version.

**Comment by harrism** [comment]
@rongou can you retarget this to 0.20?

**Comment by rongou** [comment]
@harrism done, not sure how to get the tests to pass though.

File: conda/recipes/librmm/meta.yaml
Comment (raydouglass) [inline conda/recipes/librmm/meta.yaml:38]: Why is this 1.8.2 whereas everywhere else is 1.8.5?

**Comment by raydouglass** [comment]
> not sure how to get the tests to pass though.

Looks like conflicts because `rapids-build-env` [pins spdlog to 1.7.0](https://github.com/rapidsai/integration/blob/branch-0.19/conda/recipes/versions.yaml#L145-L146).

You can follow these instructions to test: https://docs.rapids.ai/maintainers/depmgmt/#testing-a-dependency

File: conda/recipes/librmm/meta.yaml
Comment (rongou) [reply conda/recipes/librmm/meta.yaml]: Ah missed that one. Fixed.

**Comment by harrism** [comment]
@rongou do you want to submit a PR to upgrade it in the `integration` repo?

**Comment by rongou** [comment]
It was done before: https://github.com/rapidsai/integration/pull/198

**Comment by rongou** [comment]
rerun tests

**Review by harrism — APPROVED** [review]

**Review by kkraus14 — APPROVED** [review]

**Comment by rongou** [comment]
Can this be merged as is, or do I need to take out the testing changes in `ci/gpu/build.sh`?

**Comment by ajschmidt8** [comment]
> Can this be merged as is, or do I need to take out the testing changes in `ci/gpu/build.sh`?

@rongou, please remove the testing change. should be good after that

**Review by kkraus14 — APPROVED** [review]

**Comment by kkraus14** [comment]
rerun tests

**Comment by rongou** [comment]
rerun tests

**Comment by kkraus14** [comment]
@rapidsai/ops this needs a review from you, but should be good to go

**Review by ajschmidt8 — APPROVED** [review]

**Comment by ajschmidt8** [comment]
@gpucibot merge

**Pull request merged**
